src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -707,3 +743,21 @@ def calculate_expression(
     raise NotImplementedError(
         f"[Local Testing] Mocking Expression {type(exp).__name__} is not implemented."
     )
+
+
+def execute_file_operation(source_plan: MockFileOperation, analyzer: "MockAnalyzer"):
+    if source_plan.operator == MockFileOperation.Operator.PUT:
```

Comment:
Will this be hit?

Reply:
I see, the fact that eager and lazy execution shares the same codepath is very confusing 😢

Reply:
I forgot to tell you that I found in the existing snowpark-python when running against live connection,  the read/put operations are still lazy executions.. so here it's the same as the live snowpark python